### DISCUSSION TITLE
Integrate registration with GBIF for data publishing

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5846] [Feature Requests]
[**Notes and recording from our meeting with GBIF on this topic**](https://docs.google.com/document/d/1TGDJFHAnRuibvewuH4cl-nNtjv4SHAg-MG4r_Yk9u3U/edit?usp=sharing)

---

We now have a test account and organization:
https://registry.gbif-uat.org/organization/4097a450-78ca-4db3-b25a-5dc6ab7a463b

Username is `specify`, password can be obtained from @grantfitzsimmons 

---

Note that we don’t use the RSS feed at GBIF and publishers need to explicitly register the datasets they wish to see on GBIF.
 
The GBIF registry holds the following entities of interest:

1. An Installation entity to represent the Specify instance
2. The Organization entity that hosts the Specify instance (e.g. Kansas)
3. The Datasets that are published, each associated with the URL for the DwC-A file
4. The owning Organization of each Dataset, noting that one instance might publish datasets for lots of different institutions

What typically happens is the following:
 
1. An account on [GBIF.org](https://gbif.org/) is created for the Specify instance. E.g. [kansas-specify@specify.org](mailto:kansas-specify@specify.org)
2. You inform us at [helpdesk@gbif.org](mailto:helpdesk@gbif.org) that you wish to enable auto registration for an installation, and which GBIF organisations are involved in hosting the instance or publishing datasets
    a. We provide an Installation ID which specify needs to store
    b. We provide the Organisation IDs for each organization involved which specify needs to store
    c. We enable the account to manage datasets on behalf of the organisations
3. Specify can then:
   a. POST to `/dataset` to create a new dataset, storing the ID that is returned (the organization and installation IDs are used in the JSON that is posted)
   b. POST the URL for the DwC-A to `/dataset/<ID>/endpoint` to tell us where to get the data from
   c. POST to `/dataset/<id>/crawl` to trigger a crawl at any time (e.g. if you have recreated the DwC-A)

You can see an example of how this works in this script: https://github.com/gbif/registry/blob/dev/registry-examples/src/test/scripts/register.sh

**Comment by maxpatiiuk** [comment]
Is #3735 a duplicate of this?
If so, can you close one of them and merge? @grantfitzsimmons 

There is also https://github.com/specify/specify7/issues/3489, but it's probably separate
